nyx_client/nyx_client/circles.py

```diff
@@ -0,0 +1,87 @@
+# Copyright © 2024 IOTIC LABS LTD. info@iotics.com
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://github.com/Iotic-Labs/nyx-sdk/blob/main/LICENSE
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import asdict, dataclass
+from typing import Sequence
+
+
+@dataclass(frozen=True)
+class RemoteHost:
+    """Represents a remote host on the network.
+
+    Attributes:
+            name: the name of the organization
+            did: the did of the remote host
+    """
+
+    did: str
+    name: str = "Unknown host"
+
+    @classmethod
+    def from_json(cls, value: dict):
```

Comment:
(unimportant) 

1. To ref when it creates "itself":

```python
def from_json(cls, value: dict) -> "RemoteHost":
```

2. `value: dict[str, Any]`?

Reply:
:+1:  (but docstring still says json)